### PR TITLE
Implement softmax, reduction ops and squeeze

### PR DESCRIPTION
**Pull request by huningxin**
Including reduceMax, reduceMin, reduceMean, reduceProduct, reduceSum and squeeze

### CODE REVIEW

**Comment by huningxin** [comment]
@BruceDai , PTAL, thanks!

File: src/softmax.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * Compute the softmax values of the 2-D input tensor along axis 1.
+ * @param {Tensor} x
+ * @return {Tensor}
+ */
+export function softmax(x) {
+  if (x.rank !== 2) {
+    throw new Error('The input is not a 2-D tensor.');
+  }
+  const maxX = reduceMax(x, {axes: [1], keepDimensions: true});
```
Comment (BruceDai) [inline src/softmax.js:16]: There's a typos on https://github.com/webmachinelearning/webnn/blob/main/index.bs#L1901
```
    const max_x = builder.reduceMax(x, { axes: [1], keepDimensions: true });
    const exp_x = builder.exp(builder.sub(x, max));  // L1901
```
it needs update to
```
    const exp_x = builder.exp(builder.sub(x, max_x));
```

@huningxin PTAL, thanks.

**Review by BruceDai — APPROVED** [review]
LGTM, thanks @huningxin.

And I catch a typos on https://github.com/webmachinelearning/webnn, PTAL, thanks.

File: src/softmax.js (hunk 1 above)
Comment (huningxin) [reply src/softmax.js:16]: Yes, that's one. Could you please file an issue to spec?

Comment (BruceDai) [reply src/softmax.js:16]: Done, https://github.com/webmachinelearning/webnn/issues/246

Comment (huningxin) [reply src/softmax.js:16]: thanks!

**Pull request merged**